homeassistant/components/screenlogic/coordinator.py

```diff
@@ -0,0 +1,97 @@
+"""ScreenlogicDataUpdateCoordinator definition."""
+from datetime import timedelta
+import logging
+
+from screenlogicpy import ScreenLogicError, ScreenLogicGateway
+from screenlogicpy.const.common import SL_GATEWAY_IP, SL_GATEWAY_NAME, SL_GATEWAY_PORT
+from screenlogicpy.device_const.system import EQUIPMENT_FLAG
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_IP_ADDRESS, CONF_PORT, CONF_SCAN_INTERVAL
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.debounce import Debouncer
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .config_flow import async_discover_gateways_by_unique_id, name_for_mac
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUEST_REFRESH_DELAY = 2
+HEATER_COOLDOWN_DELAY = 6
+
+
+async def async_get_connect_info(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> dict[str, str | int]:
+    """Construct connect_info from configuration entry and returns it to caller."""
+    mac = entry.unique_id
+    # Attempt to rediscover gateway to follow IP changes
+    discovered_gateways = await async_discover_gateways_by_unique_id(hass)
+    if mac in discovered_gateways:
+        return discovered_gateways[mac]
+
+    _LOGGER.warning("Gateway rediscovery failed")
```

Comment:
```suggestion
    _LOGGER.debug("Gateway rediscovery failed")
```

We should tell them which one in case they have multiple